### ISSUE TITLE
[azopenai]Error when connecting to privately deployed OpenAI API endpoint via HTTP protocol

### ISSUE THREAD

**Issue by wh7f** [Azure/azure-sdk-for-go#22482]
### Bug Report

**Error Message**

`authenticated requests are not permitted for non TLS protected (https) endpoints`

**Environment**
- golang: 1.21
- SDK:
    - github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai v0.4.1
    - github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.2

**Describe**

Example source code:
```go
func BuildClient(ctx context.Context) (*azopenai.Client, error) {
	baseURL := "http://xxx" // Privately deployed OpenAI API-compatible services
	apiKey := "kkk"
	keyCredential := azcore.NewKeyCredential(apiKey)

	openaiCompatibleClient, err := azopenai.NewClientForOpenAI(baseURL, keyCredential, nil)
	if err != nil {
		return nil, err
	}

	return openaiCompatibleClient, nil
}
```

- What happened?
    - An error occurred when requesting: `authenticated requests are not permitted for non TLS protected (https) endpoints`.
- What did you expect or want to happen?
    - Send request normally.
- How can we reproduce it?
    - Turn off the restriction that `azopenai.NewClientForOpenAI` cannot use keys on http endpoints.
    - Or provide a configuration that turns off the restriction.
- Anything we should know about your environment.
    - I need to use the same SDK to access the OpenAI service provided by Azure and the independently deployed OpenAI API-compatible service.
    - The independently deployed environment is isolated from the Internet, so there will be no security issues.

**Thanks!**

`azopenai` is the best golang OpenAI API client SDK. Thank you engineers for your hard work.

**Comment by wh7f** [comment]
Yes, endpoints are provided by other teams, and they use authorization key to count calls and allocate costs.

**Comment by jhendrixMSFT** [comment]
The ability to skip the HTTPS will ship in `azcore@v1.11.0` first week of April.
